src/main/scala/org/constellation/checkpoint/CheckpointService.scala

```diff
@@ -1,33 +1,29 @@
 package org.constellation.checkpoint
 
-import cats.effect.{Concurrent, ContextShift, IO, LiftIO, Sync}
+import cats.effect.concurrent.Semaphore
+import cats.effect.{Concurrent, ContextShift, IO}
 import cats.implicits._
 import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
 import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
 import org.constellation.domain.observation.ObservationService
-import org.constellation.domain.transaction.TransactionService
-import org.constellation.p2p.{DataResolver, PeerNotification}
 import org.constellation.primitives.Schema.{CheckpointCache, _}
-import org.constellation.primitives._
-import org.constellation.storage.algebra.{Lookup, MerkleStorageAlgebra}
-import org.constellation.storage.{MessageService, NotificationService, SOEService}
+import org.constellation.primitives.Transaction
+import org.constellation.storage.algebra.Lookup
+import org.constellation.storage.{ConcurrentStorageService, MessageService, NotificationService}
 import org.constellation.{ConstellationExecutionContext, DAO}
 
 class CheckpointService[F[_]: Concurrent](
   dao: DAO,
-  transactionService: TransactionService[F],
-  messageService: MessageService[F],
-  notificationService: NotificationService[F],
-  observationService: ObservationService[F]
+  merkleService: CheckpointMerkleService[F]
 ) {
 
-  private[checkpoint] val memPool = new CheckpointBlocksMemPool[F](
-    dao,
-    transactionService.merklePool,
-    messageService.merklePool,
-    notificationService.merklePool,
-    observationService.merklePool
-  )
+  private val semaphore: Semaphore[F] = {
```

Comment:
Can you use your method for creating a semaphore?

Reply:
Yup